nats/js/kv.py

```diff
@@ -183,81 +178,9 @@ async def status(self) -> BucketStatus:
         status retrieves the status and configuration of a bucket.
         """
         info = await self._js.stream_info(self._stream)
-        kvs = KeyValue.BucketStatus()
-        kvs._nfo = info
-        kvs._bucket = self._name
-        return kvs
-
-
-class KeyValueManager(ABC):
-    """
-    KeyValueManager includes methods to create KV stores in JetStream.
-    """
-    @abstractmethod
-    async def stream_info(self, name: str):
-        pass
-
-    @abstractmethod
-    async def add_stream(self, config: api.StreamConfig):
-        pass
-
-    @abstractmethod
-    async def delete_stream(self, name: str):
-        pass
-
-    async def key_value(self, bucket: str):
-        stream = f"KV_{bucket}"
-        try:
-            si = await self.stream_info(stream)
-        except NotFoundError:
-            raise BucketNotFoundError
-        if si.config.max_msgs_per_subject < 1:
-            raise BadBucketError
-
-        return KeyValue(
-            name=bucket,
-            stream=stream,
-            pre=f"$KV.{bucket}.",
-            js=self,
-        )
-
-    async def create_key_value(self, **params) -> KeyValue:
-        """
-        create_key_value takes an api.KeyValueConfig and creates a KV in JetStream.
-        """
-        config = api.KeyValueConfig.loads(**params)
-        if not config.history:
-            config.history = 1
-        if not config.replicas:
-            config.replicas = 1
-        if config.ttl:
-            config.ttl = config.ttl * 1_000_000_000
-
-        stream = api.StreamConfig(
-            name=f"KV_{config.bucket}",
-            description=None,
-            subjects=[f"$KV.{config.bucket}.>"],
-            max_msgs_per_subject=config.history,
-            max_bytes=config.max_bytes,
-            max_age=config.ttl,
-            max_msg_size=config.max_value_size,
-            storage=config.storage,
-            num_replicas=config.replicas,
-            allow_rollup_hdrs=True,
-            deny_delete=True,
-        )
-        await self.add_stream(stream)
+        return KeyValue.BucketStatus(stream_info=info, bucket=self._name)
 
-        return KeyValue(
-            name=config.bucket,
-            stream=stream.name,
-            pre=f"$KV.{config.bucket}.",
-            js=self,
-        )
-
-    async def delete_key_value(self, bucket: str):
-        """
-        delete_key_value deletes a JetStream KeyValue store by destroying
-        the associated stream.
+    async def destroy(self) -> bool:
```

Comment:
I think this is a good addition but we should also let `delete_key_value` to more or less conform to the APIs in the Go client just in case: https://github.com/nats-io/nats.go/blob/b337a5c01afdab41a22becdf3759e038f9948889/kv.go#L35

Reply:
My motivation to change it was that `delete_key_value` didn't work and I wasn't sure how it was supposed to 🙃 Thank you for pointing to the Go client, I'll just have a look at it and do the same here.